skcosmo/sparse_methods/SparseKPCA.py

```diff
@@ -0,0 +1,266 @@
+import numpy as np
+from functools import partial
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+from sklearn.utils import check_array
+from sklearn.exceptions import NotFittedError
+from sklearn.metrics.pairwise import pairwise_kernels
+
+from ..utils import eig_solver
+from ..selection.FPS import SampleFPS
+from ..preprocessing.flexible_scaler import SparseKernelCenterer
+
+
+class SparseKPCA(TransformerMixin, BaseEstimator):
+    """SparseKernel Principal component analysis (SKPCA).
+
+    Non-linear dimensionality reduction through the use of sparse kernels
+    determined via the Nystrom approximation.
+
+    Parameters
+    ----------
+
+    n_active : int
+        Number of active samples to use within the sparse kernel.
+
+    n_components : int, default=n_active
+        Number of components.
+
+    kernel : {'linear', 'poly', \
+            'rbf', 'sigmoid', 'cosine', 'precomputed'}, default='linear'
+        Kernel used for PCA.
+
+    gamma : float, default=None
+        Kernel coefficient for rbf, poly and sigmoid kernels. Ignored by other
+        kernels. If ``gamma`` is ``None``, then it is set to ``1/n_features``.
+
+    degree : int, default=3
+        Degree for poly kernels. Ignored by other kernels.
+
+    coef0 : float, default=1
+        Independent term in poly and sigmoid kernels.
+        Ignored by other kernels.
+
+    kernel_params : dict, default=None
+        Parameters (keyword arguments) and
+        values for kernel passed as callable object.
+        Ignored by other kernels.
+
+    alpha : float, default=1.0
+        Hyperparameter of the ridge regression that learns the
+        inverse transform (when fit_inverse_transform=True).
+
+    fit_inverse_transform : bool, default=False
+        Learn the inverse transform for non-precomputed kernels.
+        (i.e. learn to find the pre-image of a point)
+
+    tol : float, default=0
+        Convergence tolerance for arpack.
+        If 0, optimal value will be chosen by arpack.
+
+    copy_X : bool, default=True
+        If True, input X is copied and stored by the model in the `X_fit_`
+        attribute. If no further changes will be done to X, setting
+        `copy_X=False` saves memory by storing a reference.
+
+    n_jobs : int, default=None
+        The number of parallel jobs to run.
+        ``None`` means 1 unless in a :obj:`joblib.parallel_backend` context.
+        ``-1`` means using all processors. See :term:`Glossary <n_jobs>`
+        for more details.
+
+    """
+
+    def __init__(
+        self,
+        n_active,
+        n_components=None,
+        kernel="linear",
+        gamma=None,
+        degree=3,
+        coef0=1,
+        kernel_params=None,
+        selector=SampleFPS,
+        center=False,
```

Comment:
Would it be better to have centering as true by default?
We also need a separate argument for scaling if we separate the centering from the scaling

```suggestion
        center=True,
```